crates/chain-state/src/deferred_trie.rs

```diff
@@ -0,0 +1,295 @@
+use alloy_primitives::B256;
+use reth_trie::{updates::TrieUpdatesSorted, HashedPostStateSorted, TrieInputSorted};
+use std::{
+    fmt,
+    sync::{Arc, OnceLock},
+};
+
+/// Sorted trie data computed for an executed block.
+/// These represent the complete set of sorted trie data required to persist
+/// block state for, and generate proofs on top of a block.
+#[derive(Clone, Debug, Default)]
+pub struct ComputedTrieData {
+    /// Sorted hashed post-state produced by execution.
+    pub hashed_state: Arc<HashedPostStateSorted>,
+    /// Sorted trie updates produced by state root computation.
+    pub trie_updates: Arc<TrieUpdatesSorted>,
+    /// The persisted ancestor hash this trie input is anchored to.
+    pub anchor_hash: Option<B256>,
```

Comment:
I'd like to put these in the same Option, the TrieInputSorted is not valid without knowing what its anchor hash is, so we don't want to allow the possibility that it won't be present.